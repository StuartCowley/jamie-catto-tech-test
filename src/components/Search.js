import React, { useState } from 'react';
import getImage from '../requests/getImage';
import '../styles/search.css';
import PropTypes from 'prop-types';

const Search = ({setSearchResults}) => {
    const [value, setValue] = useState();

    const handleSubmit = async (event) => {
        event.preventDefault();
        const images = await getImage(value);
        console.log(images);
        setSearchResults(images);
    }
    return (
        <>
        <form className="search-form" onSubmit={handleSubmit}>
            <input type="text" className="search-box"
            onChange={(e) => setValue(e.target.value)} />
            <button 
            className="search-btn"
            type="submit"
            data-testid="search-btn">Search</button>
        </form>
        </>
    )
}

Search.propTypes = {
    setSearchResults: PropTypes.func,
}

export default Search;