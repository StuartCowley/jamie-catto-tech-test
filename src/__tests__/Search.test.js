import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const validProps = jest.fn();
  
  it("renders correctly", () => {
    const { asFragment } = render(<Search />);
    
    expect(asFragment()).toMatchSnapshot();
  });
    
  it('displays a search button with correct text', () => {
    render(<Search setSearchResults={validProps} />);

    expect(screen.getAllByRole('button')).toHaveLength(1);
    expect(screen.getByTestId('search-btn')).toHaveTextContent('Search');
  });

  it('calls the right function when clicked', async () => {
    render(<Search setSearchResults={validProps} />)

    await fireEvent.click(screen.getByRole("button"));

    expect(validProps).toHaveBeenCalled();
  })

});