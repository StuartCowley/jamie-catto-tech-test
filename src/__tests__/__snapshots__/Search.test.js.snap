// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-box"
      type="text"
    />
    <button
      class="search-btn"
      data-testid="search-btn"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
